hack/verify-examples-kind.sh

```diff
@@ -106,6 +105,9 @@ for CHANNEL in experimental standard; do
         echo Examples failed as expected
 done
 
+# Undo workaround from earlier
```

Comment:
Ooh, this is a good catch.

Reply:
"Temporary workaround for release" sounds like something that should just be removed? If it's needed permanently, maybe we need to update the comment?

Reply:
I think you're right, this probably shouldn't exist at all. There was a time where we wanted the manifests installed from the main branch to be at least somewhat useful and refer to the latest released image instead of the latest image build from the main branch. All of our installation instructions clearly refer to a versioned release so this is likely entirely unnecessary, will revert this change altogether.

Reply:
Actually as I was digging into this more, it looks more complicated than initially expected and I don't want to expand this PR any further. I've created https://github.com/kubernetes-sigs/gateway-api/issues/1366 to track this and added it to the v0.6.0 milestone.